map/bookmark_manager.cpp

```diff
@@ -172,10 +182,17 @@ BookmarkManager::SharingResult ExportMultipleFiles(BookmarkManager::KMLDataColle
   return {std::move(categoriesIds), std::move(kmzFilePath)};
 }
 
-BookmarkManager::SharingResult GetFileForSharing(BookmarkManager::KMLDataCollectionPtr collection)
+BookmarkManager::SharingResult GetFileForSharing(BookmarkManager::KMLDataCollectionPtr collection, KmlFileType kmlFileType)
 {
   if (collection->size() == 1)
-    return ExportSingleFile(collection->front());
+    switch (kmlFileType)
+    {
+    case KmlFileType::Text: return ExportSingleFileKml(collection->front());
+    case KmlFileType::Gpx: return ExportSingleFileGpx(collection->front());
+    default:
+      LOG(LERROR, ("Unexpected file type", kmlFileType));
+      return {{collection->front().second->m_categoryData.m_id}, BookmarkManager::SharingResult::Code::FileError,"Unexpected file type"};
+    }
   else
     return ExportMultipleFiles(collection);
```

Comment:
Sorry, didn't see that there are no bracers here in this switch, it looks strange...
Would it be clearer to rewrite it like this?
```
if (collection->size() > 1)
  return ExportMultipleFiles(collection);

switch (kmlFileType)
...
```

Reply:
Fixed